UI-Router V1 Support

iaculch opened the pull request:
After migrating to ui-router version 1.x the restrict keyword defined on state objects stopped working resulting in a JS error when navigating to restricted routes.

A change has been made in the way that State Change Events are fired in the new version: https://ui-router.github.io/guide/ng1/migrate-to-1_0#state-change-events

This patch checks for the new $transitions service to determine the version of ui-router and binds the appropriate state change mechanism.

iaculch committed:
Use new Transition Hook API provided by UI-Router V1

https://ui-router.github.io/guide/ng1/migrate-to-1_0#state-change-events

Update release files

Support UI-Router 0.x and 1.x

alonbardavid commented:
I'm a bit busy at the moment, so I'll try and get to this as soon as I can.

But before that, any chance you can get rid of the old functionality rather then check if uiRouterV1 exists? I'd rather just mark a new version as requiring a higher version of ui-router then keeping multiple version in the same db.

alonbardavid commented on release/visor.js:
```diff
@@ -1094,77 +1094,120 @@ if (typeof module !== 'undefined' && typeof exports !== 'undefined' && module.ex
+            try {
+                $injector.get('$state');
+                uiModuleExists = true;
+                $injector.get('$transitions');
```
if $transition doesn't exists, this will fail before the function is even ran, since you added the $transitions in the function arguments injection

iaculch replied:
Will remove this check in favour of a version which only supports UI-Router v1+

iaculch commented:
Updated, now assuming ui-router version >1.0

iaculch committed:
fix(ui-router-angular-hybrid): remove $timeout wrapper around notAllowed invocation

This $timeout workaround is no longer necessary now that $transitions service is used
instead of $stateChangeStart

iaculch commented:
Added fix for #19

sambernet commented:
@iaculch why exactly did you close the PR here? Did you just give up on getting this merged? 

I'm asking because we're also using your branch and I still have hope to get away from the `git:`-style dependency in our `package.json` files, back to a regular release version 🤞
